src/ripple/core/impl/DeadlineTimer.cpp

```diff
@@ -229,6 +297,7 @@ class DeadlineTimer::Manager
     std::condition_variable_any wakeup_;  // Works with std::recursive_mutex.
     std::thread thread_;
     bool shouldExit_ {false};
+    int recursionCount_ {0};
```

Comment:
Nice.  I assume you''re running this well past sync?  After looking more into the uses of `DeadlineTimer` I have confidence that the `LogicError` will not fire.  But having it programmatically checked as you've done will help me sleep. :-)